src/Adapter/Product/SpecificPrice/Repository/SpecificPriceRepository.php

```diff
@@ -160,6 +172,82 @@ public function getProductSpecificPricesCount(ProductId $productId, ?array $filt
         return (int) $qb->execute()->fetch()['total_specific_prices'];
     }
 
+    /**
+     * @param ProductId $productId
+     * @param PriorityList $priorityList
+     */
+    public function setPrioritiesForProduct(ProductId $productId, PriorityList $priorityList): void
+    {
+        $exception = new CoreException(sprintf(
+            'Error occurred when trying to set specific price priorities for product #%d',
+            $productId->getValue()
+        ));
+
+        try {
+            if (!SpecificPrice::setSpecificPriority($productId->getValue(), $priorityList->getPriorities())) {
```

Comment:
> Alternatively, since there is no hook in the legacy method, you could perform the query using $this->connection just like you did in removePrioritiesForProduct

It would be duplicating logic just so it looks better. Not sure if its worth it now :thinking: